adr/0004-application-architecture.md

```diff
@@ -0,0 +1,42 @@
+# 4. Application architecture
+
+Date: 2019-09-25
+
+## Status
+
+Accepted
+
+## Context
+
+We are starting to build a beta of the 'Apply for postgraduate teacher training' service. This service will have 5 distinct user groups, who will have their own interface in the service.
+
+1. Candidates are people who would like to become a teacher.
+2. Provider users work for an organisation that provides teacher training, like a school or university
+3. Vendors make software that universities use for administration
+4. Referees will be asked by candidates to provide a reference
```

Comment:
```suggestion
4. Referees will be provided by candidates and the service will ask them to provide a reference
```

I might be wrong, but I believe this is more accurate. The `asked by candidates` suggests it's a thing that a candidate will have to do but from my understanding it's a thing we'll handle.